### PR TITLE
#3 Adding Facebook Auth

### PR DESCRIPTION
**Pull request by johnolos**
work in progress -- comment if you'd like

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,8 +29,10 @@
     "jest": "^18.1.0",
     "jest-cli": "^18.1.0",
     "mongoose": "^4.7.7",
+    "node-yaml-config": "^0.0.4",
```
Comment (hawkaa) [inline package.json:33]: Awesome!

File: src/db/models/user.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import mongoose from 'mongoose';
+import passportLocalMongoose from 'passport-local-mongoose';
+
+const Schema = mongoose.Schema;
+
+const User = new Schema({
+  name: { type: String, required: true },
```
Comment (hawkaa) [inline src/db/models/user.js:7]: Split into multiline?

File: src/index.js [hunk 3]
```diff
@@ -1,31 +1,81 @@
...
+
+const config = yamlConfig.load(path.join(__dirname, '/config.yml'));
+
+const mongodbUrl = `mongodb://${config.database.host}:${config.database.port}/${config.database.db}`;
+
+winston.log(`Connecting to mongodb at: ${mongodbUrl}`);
+
+mongoose.connect(mongodbUrl);
+
+winston.log('info', config);
```
Comment (hawkaa) [inline src/index.js:20]: 💯

File: src/index.js [hunk 4]
```diff
@@ -1,31 +1,81 @@
...
+passport.use(new FacebookStrategy({
+  clientID: config.facebook.appId,
+  clientSecret: config.facebook.secret,
+  callbackURL: config.facebook.callbackUrl,
+  display: 'popup',
+  authType: 'code',
+  passReqToCallback: 'true',
+  enableProof: true,
+}), (request, accessToken, refreshToken, profile, callback) => {
+  User.findOne({ oauthId: profile.id }, (error, user) => {
```
Comment (hawkaa) [inline src/index.js]: Change this to "promise pattern" with `then` and `error` http://mongoosejs.com/docs/promises.html. We should also consider adding Bluebird.

**Comment by hawkaa** [comment]
Good stuff, @johnolos  💃

File: src/index.js (hunk 4 above)
Comment (johnolos) [reply src/index.js]: :+1:  for bluebird

File: src/db/models/user.js (hunk 2 above)
Comment (johnolos) [reply src/db/models/user.js:7]: I think it reads better with columns separated by newline regardless of properties of said column.

Comment (hawkaa) [reply src/db/models/user.js:7]: 👍

**Review by hawkaa — APPROVED** [review]
LGTM 👍

**Pull request merged**